### PR TITLE
Codegen an entrypoint for flutter web applications

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Generate an entrypoint that is flutter web specific. This avoids needing to reference the "webonly" members for now. ~~ Still need to wire up dart2js with something similar.~~ Done

cc @hterkelsen, I imagine you would use this for plugins too

### CODE REVIEW

File: examples/flutter_gallery/lib/main_web.dart [hunk 1]
```diff
@@ -1,16 +0,0 @@
-// Copyright 2019 The Chromium Authors. All rights reserved.
```
Comment (jonahwilliams) [inline examples/flutter_gallery/lib/main_web.dart]: This is no longer needed, hurrah

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 2]
```diff
@@ -254,7 +290,7 @@ class FlutterWebEntrypointBuilder implements Builder {
   Future<void> build(BuildStep buildStep) async {
     bool matches = false;
     for (String target in targets) {
-      if (buildStep.inputId.path.contains(target)) {
+      if (buildStep.inputId.path.contains(fs.path.setExtension(target, '_web_entrypoint.dart'))) {
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Would this not result in `blah._web_entrypoint.dart`?

File: packages/flutter_tools/lib/src/web/compile.dart [hunk 3]
```diff
@@ -4,80 +4,56 @@
...
+import '../base/logger.dart';
 import '../build_info.dart';
-import '../convert.dart';
+import '../bundle.dart';
 import '../globals.dart';
+import '../project.dart';
 
-/// The [WebCompiler] instance.
-WebCompiler get webCompiler => context.get<WebCompiler>();
+/// The [WebCompilationProxuy] instance.
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/web/compile.dart]: typo

**Review by yjbanov — APPROVED** [review]
LGTM if LGT @hterkelsen

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart [hunk 4]
```diff
@@ -195,19 +226,28 @@ class BuildRunnerWebCompilationProxy extends WebCompilationProxy {
...
+      result = await _runBuilder(
+        buildEnvironment,
         buildOptions,
+        targets,
+        release,
+        buildDirs,
+      );
+      return result.status == core.BuildStatus.success;
+    } on core.BuildConfigChangedException {
+      projectDirectory
```
Comment (jakemac53) [inline packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Ideally you wouldn't delete the entire `.dart_tool` dir - pub also uses it to cache local dependency snapshots for binaries.

You could delete `.dart_tool/build` but that would also bash over any other builds.

We do have the functionality to do a more fine grained cleanup, that only deletes the actual known outputs as well, see here https://github.com/dart-lang/build/blob/master/build_runner/lib/src/entrypoint/clean.dart#L41.

**Review by harryterkelsen — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Surprisingly not

File: packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart]: Awesome,I tried to adapt that code to this in a way that makes sense

**Pull request merged**